src/text-overflow/TextOverflow.d.ts

```diff
@@ -0,0 +1,16 @@
+import { PureComponent } from 'react';
+
+declare namespace TextOverflow {
+  type Props = {
+    label?: string | number,
+    popover?: object, // TODO(typo): Omit?
+    lazy?: boolean,
+    id?: string,
+    className?: string,
+    style?: object
+  }
```

Comment:
you should reflect exactly the `Props` object from `TextOverflow.js`

here you added `popover` while `children` and `delayWhenLazy` are both missing.